plugins/useractivity/useractivity.go

```diff
@@ -36,11 +36,16 @@ type reporter struct {
 }
 
 type session struct {
-	start      time.Time
-	lastAct    time.Time
-	id         string
-	activities map[string]bool
-	clients    map[string]bool
+	start         time.Time
+	lastAct       time.Time
+	id            string
+	activityCount int64
+	activities    map[string]bool
+	clients       map[string]bool
+	projects      map[string]bool
+
+	// This mutex synchronizes access to all fields.
+	mut sync.Mutex
```

Comment:
Now that a session struct contains a sync.Mutex, the identity of session objects is meaningful. Before, all `session` fields were either values (time.Time, string) or pointers to mutable data structures (map[string]bool). Treating session with "value-like" semantics works for both of these.

It doesn't work for a mutex. The Lock/Unlock operations at e.g. 189-190 will only apply to the mutex inside the local session variable.

So, we must do one of two things:

1. Change the r.sessions map to store `*session`, rather than `session`.
2. Change `mut` to a `*sync.Mutex`, and allocate a shared mutex when we construct a `session`.

The choice between the two is a bit subtle, but I think (1) is somewhat more straightforward: it is easier to think about the session as being an object that owns all its members, rather than a struct that contains some local values and some pointers to state shared among all session values with the same key.